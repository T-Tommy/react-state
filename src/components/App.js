import React from 'react';
import ColorPicker from './ColorPicker';

export default function App() {
  return (
    <ColorPicker colors={ ['Red', 'Orange', 'Yellow', 'Green', 'Blue', 'Purple'] } />
  );
}
