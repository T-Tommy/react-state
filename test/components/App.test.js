import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/components/App';

describe('App component', () => {
  it('renders an App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an App', () => {
    const wrapper = shallow(<App />);

    wrapper.instance().updateColor('Purple');
    expect(wrapper.state('color')).toEqual('Purple');
  });
});
