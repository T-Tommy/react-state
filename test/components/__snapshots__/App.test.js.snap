// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders an App 1`] = `
<Fragment>
  <ColorPicker
    colors={
      Array [
        "Purple",
        "Blue",
        "Green",
        "Yellow",
        "Orange",
        "Red",
      ]
    }
    method={[Function]}
  />
  <ColorDisplay
    color="black"
  />
</Fragment>
`;
